services/search/utils.py

```diff
@@ -11,15 +11,35 @@ def create_synonym_list_esp(word_arr, max_synonyms=5):
             optional: Default value 5
     Returns: a list of synonyms
     """
+    session = FuturesSession()
+    reqs = create_conc_reqs(word_arr, session)
+
     synonyms = []
-    for word in word_arr:
-        payload = requests.get(constants.SPANISH_API_URL+word)
-        payload.raise_for_status()
 
-        word_synonyms = payload.json()['sinonimos']
+    for req in reqs:
+        resp = req.result()
+
+        word_synonyms = resp.json()['sinonimos']
+
         if len(word_synonyms) > max_synonyms:
             word_synonyms = word_synonyms[:max_synonyms]
         for synonym in word_synonyms:
             synonyms.append(synonym['sinonimo'])
 
     return synonyms
+
+
+def create_conc_reqs(word_arr, session):
+    """
+    Create asynchronous requests list using the futures session
+    Attributes:
+        word_arr: list of all requests to be made
+        session: requests_futures instanced session
+    Return:
+        Asynchronous HTTP request list
+    """
+    reqs = []
+    for word in word_arr:
+        reqs.append(session.get(constants.SPANISH_API_URL+word))  # pylint: disable=no-member
```

Comment:
Does the `Futures` library has a limit for concurrent requests that we should care about? Imagine if you go through a `word_arr` containing 1,000,000 words, would the process run out of memory? Or could it handle it?

Reply:
I think there's not a limit. There's a max workers limit set in the constructor which specifies the number of threads, I believe if we exceed it we will just do a 2nd, 3rd or nth cycle to fulfill all our requests. Having said that do you believe we should set a hard limit of the input query on the front end? I think a user shouldn't even exceed even a 100 word search query let alone a million one.